### PR TITLE
Fix checkbox rendering and clean up build config

### PR DESCRIPTION
**Pull request by tiltec**
## What does this PR do?

`process.env.THEME` was missing, but quasar was depending on it.

I took the chance to clean up our build config a bit more.

- `__THEME` option is not really needed, it seems fine to hardcode `mat`
- Import only CSS from `quasar-extras` and not the whole directory (that's what quasar-cli does, but maybe it's also no difference)
- Make `/config` into `/build/config.js`
- Move process.env definitions to `config.js`
- Move alias definitions to webpack config

## Checklist

- [ ] added a test, or explain why one is not needed/possible...
- [ ] no unrelated changes
- [x] asked someone for a code review
- [x] joined #karrot-dev channel at https://slackin.yunity.org
- [ ] added an entry to CHANGELOG.md (description, pull request link, username(s))

### CODE REVIEW

File: build/config.js [hunk 1]
```diff
@@ -25,29 +23,28 @@ const backendProxy = {
...
-    '>': path.resolve(__dirname, '../test'),
-    variables: path.resolve(__dirname, '../src/themes/quasar.variables.styl'),
-    slidetoggle: path.resolve(__dirname, '../src/themes/karrot.slidetoggle.styl'),
-    editbox: path.resolve(__dirname, '../src/themes/karrot.editbox.styl'),
-  },
+const theme = require('./env-utils').platform.theme
 
-  // Backend to make API requests to
+const env = {
+  NODE_ENV: '"production"',
```
Comment (nicksellen) [inline build/config.js]: Could use consistent `JSON.stringify()` for values

File: build/webpack.config.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ module.exports = {
   },
   output: {
     path: resolve(__dirname, '../dist'),
-    publicPath: config[env.prod ? 'build' : 'dev'].publicPath,
+    publicPath: env.prod ? '' : '/',
```
Comment (nicksellen) [inline build/webpack.config.js]: I'm not sure we actually need this difference in dev/prod

File: build/webpack.config.js [hunk 3]
```diff
@@ -53,7 +53,15 @@ module.exports = {
...
     ],
-    alias: config.aliases,
+    alias: {
+      quasar: 'quasar-framework',
+      'quasar-vue-plugin': 'quasar-framework/src/vue-plugin',
+      '@': resolve(__dirname, '../src'),
+      '>': resolve(__dirname, '../test'),
+      variables: resolve(__dirname, '../src/themes/quasar.variables.styl'),
+      slidetoggle: resolve(__dirname, '../src/themes/karrot.slidetoggle.styl'),
+      editbox: resolve(__dirname, '../src/themes/karrot.editbox.styl'),
```
Comment (nicksellen) [inline build/webpack.config.js]: I know these were here before, but doesn't seem quite the right things to be aliases...

File: build/config.js [hunk 4]
```diff
@@ -25,29 +23,28 @@ const backendProxy = {
...
-  aliases: {
-    quasar: 'quasar-framework',
-    'quasar-vue-plugin': 'quasar-framework/src/vue-plugin',
-    '@': path.resolve(__dirname, '../src'),
-    '>': path.resolve(__dirname, '../test'),
-    variables: path.resolve(__dirname, '../src/themes/quasar.variables.styl'),
-    slidetoggle: path.resolve(__dirname, '../src/themes/karrot.slidetoggle.styl'),
-    editbox: path.resolve(__dirname, '../src/themes/karrot.editbox.styl'),
-  },
+const theme = require('./env-utils').platform.theme
```
Comment (nicksellen) [inline build/config.js]: Do we even still need env-utils?

**Comment by tiltec** [comment]
Thanks! I'll merge this and create a follow-up PR with further improvements.

**Pull request merged**